### PR TITLE
added token call to get available regions.

### PR DESCRIPTION
**Pull request by ashleyschuett**
@normanjoyner 

I need this to updated rackspace in providers.

### CODE REVIEW

File: lib/pkgcloud/rackspace/client.js [hunk 1]
```diff
@@ -33,3 +33,24 @@ Client.prototype._getIdentityOptions = function() {
...
+          auth : {
+              "RAX-KSKEY:apiKeyCredentials": {
+                  "username":this.config.username,
+                  "apiKey":this.config.apiKey
+              }
+          }
+      }
+    };
+
+    this._request(options, function (err, body) {
```
Comment (nicktate) [inline lib/pkgcloud/rackspace/client.js]: you can just do `this._request(options, callback);`

File: lib/pkgcloud/rackspace/client.js [hunk 2]
```diff
@@ -33,3 +33,24 @@ Client.prototype._getIdentityOptions = function() {
   }, Client.super_.prototype._getIdentityOptions.call(this));
 };
 
+Client.prototype.getToken = function (callback) {
+    var options = {
+      uri: this.config.authUrl}+'/v2.0/tokens',
+      method: 'POST',
+      body: {
+          auth : {
+              "RAX-KSKEY:apiKeyCredentials": {
```
Comment (nicktate) [inline lib/pkgcloud/rackspace/client.js]: maybe include a link to docs about what this key is

File: lib/pkgcloud/rackspace/client.js [hunk 3]
```diff
@@ -33,3 +33,24 @@ Client.prototype._getIdentityOptions = function() {
   }, Client.super_.prototype._getIdentityOptions.call(this));
 };
 
+Client.prototype.getToken = function (callback) {
+    var options = {
+      uri: this.config.authUrl}+'/v2.0/tokens',
```
Comment (nicktate) [inline lib/pkgcloud/rackspace/client.js]: erroneous `}` curly in the `uri` value

**Comment by ashleyschuett** [comment]
@nicktate updated

**Pull request merged**